x-pack/plugins/metrics_entities/server/routes/delete_transforms.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { IRouter } from '../../../../../src/core/server';
+import { METRICS_ENTITIES_TRANSFORMS } from '../../common/constants';
+import { ModuleNames } from '../modules';
+
+import { getMetricsEntitiesClient } from './utils/get_metrics_entities_client';
+
+/**
+ * Deletes transforms.
+ * NOTE: We use a POST rather than a DELETE on purpose here to ensure that there
+ * are not problems with the body being sent.
+ * @param router The router to delete the collection of transforms
+ */
+export const deleteTransforms = (router: IRouter): void => {
```

Comment:
@elastic/ml-ui 

I wonder if there is room for consolidation. We discussed providing kibana API's for things like this.

Reply:
Could be, yeah. I will swap out the API's I have now for the kibana ones as suggested below by @walterra and re-push up and we can see what's left.